packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -102,12 +102,24 @@ if (__DEV__) {
   const NORMALIZE_NEWLINES_REGEX = /\r\n?/g;
   const NORMALIZE_NULL_AND_REPLACEMENT_REGEX = /\u0000|\uFFFD/g;
 
-  normalizeMarkupForTextOrAttribute = function(markup: mixed): string {
+  normalizeMarkupForTextOrAttribute = function(
+    markup: mixed,
+    propName?: string,
+  ): string {
     const markupString =
       typeof markup === 'string' ? markup : '' + (markup: any);
-    return markupString
+    const markupNormalized = markupString
       .replace(NORMALIZE_NEWLINES_REGEX, '\n')
       .replace(NORMALIZE_NULL_AND_REPLACEMENT_REGEX, '');
+    if (propName !== 'style') {
+      return markupNormalized;
+    }
+    // IE 11 parses & normalizes the style attribute as opposed to other
+    // browsers. The following normalization will still fail if the style
+    // attribute contains invalid CSS declarations but the majority of
+    // false warnings comes from spacing issues.
+    // See https://github.com/facebook/react/issues/11807
+    return markupNormalized.replace(/\s*([:;])\s*/, '$1').replace(/;$/, '');
```

Comment:
How confident are you in this? What if `;` is part of an image URL or something?

Reply:
That's an excellent point. I'm afraid the only certain solution would be to do CSS parsing of the declarations included in the `style` attribute but the grammar for that is not that simple:
https://www.w3.org/TR/css-syntax-3/#tokenizing-and-parsing
especially around unquoted URLs so this might be too much.

That said, spaces are not that frequent in URLs, especially around semicolons and colons so this change should have very few false positives as opposed to having false negatives for every `style` attribute.

If I understand correctly, this only affects development. How about keeping this logic but only if `document.documentMode` is truthy? Regular browsers would work as they are now and developing in IE would be a little less unpleasant.

Reply:
I guess it's fine since we normalize it on both sides. So at most you'll miss some legitimate warnings, but you shouldn't get warnings about valid code. Is that accurate?

Your suggestion to gate this by `document.documentMode` sounds good to me too.

Reply:
> So at most you'll miss some legitimate warnings, but you shouldn't get warnings about valid code. Is that accurate?

Yes, that's exactly it. And in non-IE browsers you'll still get all the warnings.